### PR TITLE
Azure Linux Builds

### PR DESCRIPTION
**Pull request by dgasmith**
## Description
The plan is to switch away from TravisCI Linux builds to Azure for unlimited build minutes, the ability to host more complex pipelines, and to prevent splitting between multiple services.

## Status
- [x] Ready for review
- [x] Ready for merge

### CODE REVIEW

File: .azure-pipelines/azure-pipelines-linux.yml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+                    pycppe ^
+                    pylibefp ^
+                    snsmp2 ^
+                    mp2d ^
+                    pytest-xdist ^
+                    -c psi4/label/dev
+       conda install qcelemental ^
+                     qcengine ^
+                     pybind11 ^
+                     -c conda-forge
```
Comment (loriab) [inline .azure-pipelines/azure-pipelines-linux.yml]: why c-f not psi4/label/dev?

File: .azure-pipelines/azure-pipelines-linux.yml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  - script: |
+      export PATH="$HOME/miniconda/bin:$PATH"
+      python ../.scripts/travis_run_test.py
+      python ../.scripts/travis_print_failing.py
+    displayName: 'CMake Test'
+
+  - script: |
+      export PATH="$HOME/miniconda/bin:$PATH"
+      PYTHONPATH=stage/lib/ pytest -v -rws --durations=15 -n 2 -m 'quick' stage/lib/psi4/tests/
+    displayName: 'Python Test'
```
Comment (loriab) [inline .azure-pipelines/azure-pipelines-linux.yml]: "Pytest Test"

File: .azure-pipelines/azure-pipelines-linux.yml [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  - script: |
+      export PATH="$HOME/miniconda/bin:$PATH"
+      ./stage/bin/psi4 ../tests/tu1-h2o-energy/input.dat
+    displayName: 'Spot Test'
+
+  - script: |
+      export PATH="$HOME/miniconda/bin:$PATH"
+      python ../.scripts/travis_run_test.py
+      python ../.scripts/travis_print_failing.py
+    displayName: 'CMake Test'
```
Comment (loriab) [inline .azure-pipelines/azure-pipelines-linux.yml]: "CTest Test"

File: .azure-pipelines/azure-pipelines-linux.yml [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+        gcp \
+        resp \
+        pycppe \
+        pylibefp \
+        snsmp2 \
+        mp2d \
+        pytest-xdist \
+        -c psi4/label/dev
+      conda install \
+        qcelemental \
```
Comment (robertodr) [inline .azure-pipelines/azure-pipelines-linux.yml]: I think this needs `blas=*=mkl` and `mkl-include` too.

File: .azure-pipelines/azure-pipelines-linux.yml (hunk 1 above)
Comment (dgasmith) [reply .azure-pipelines/azure-pipelines-linux.yml]: I pulled this from Travis.

File: .azure-pipelines/azure-pipelines-linux.yml (hunk 4 above)
Comment (dgasmith) [reply .azure-pipelines/azure-pipelines-linux.yml]: BLAS should come from NumPy, but doesn't hurt to be explicit.

Comment (robertodr) [reply .azure-pipelines/azure-pipelines-linux.yml]: Travis was using BLAS/LAPACK from `liblapack-dev`. You might need to export `MATH_ROOT` too.

File: .azure-pipelines/azure-pipelines-linux.yml (hunk 1 above)
Comment (loriab) [reply .azure-pipelines/azure-pipelines-linux.yml]: figure out why it's from c-f. this makes sure psi is using latest, even if I'm behind on psi4-dev packages.

**Comment by dgasmith** [comment]
@psi4/admins Ready to review!

**Comment by robertodr** [comment]
I'll close #1748, but is the plan to use Ninja also on Linux/macOS?

**Comment by dgasmith** [comment]
It didn't seem to help much on Travis or am I wrong there? Very on board with ninja overall, just balancing the complexity.

**Comment by robertodr** [comment]
That's correct, not much help on Travis. I was under the impression it was going to be a requirement when moving to libint2. But I can have a go at it later, once this is merged.

**Comment by robertodr** [comment]
Someone should let Travis know they're not required anymore :smiley_cat:

**Comment by dgasmith** [comment]
Travis and GitHub has been informed!

**Pull request merged**